cli/desbordante.py

```diff
@@ -0,0 +1,319 @@
+        return result
+    except Exception as exc:
+        click.echo(exc)
+        sys.exit(1)
+
+
+def create_verbose_info(not_omitted_opts, start_point, end_point):
+    info = ''
+    for name, value in not_omitted_opts.items():
+        info = info + str(name) + ': ' + str(value) + '\n'
+    info = (info + 'Elapsed time during execution in seconds: ' +
+            str(end_point - start_point) + '\n\nResult:\n')
+    return info
+
+
+def print_result(result, filename, verbose_info):
+    if verbose_info is not None:
+        result = verbose_info + str(result)
+    try:
+        if filename is not None:
+            with open(filename, "w") as file:
+                print(result, file=file)
+        else:
+            click.echo(result)
+    except Exception as exc:
+        click.echo(exc)
+        sys.exit(1)
+
+
+def print_unused_opts(used_opts, not_omitted_opts):
+    used_opts.extend(['task', 'algo', 'separator', 'has_header', 'verbose'])
+
+    for opt in used_opts:
+        not_omitted_opts.remove(opt)
+
+    if not_omitted_opts:
+        click.echo('Unused options: ', nl=False)
+        click.echo(not_omitted_opts)
+
+
+def print_help_page(algo_name, task):
+    if algo_name is not None:
+        click.echo(ALGO_HELP_PAGES[algo_name])
+    elif task is not None:
+        click.echo(TASK_HELP_PAGES[task])
+    else:
+        click.echo(HELP)
+
+
+def return_not_omitted_options(args):
+    not_omitted_opts = {}
+    for name, value in args.items():
+        if not is_omitted(value):
+            not_omitted_opts.update({name: value})
+    return not_omitted_opts
+
+
+def create_and_fill_opts():
+    opts = {}
+    for algo_type in ALGOS.values():
+        algo = algo_type()
+        for opt in algo.get_possible_options():
+            opts.update({opt: algo.get_option_type(opt)})
+    return opts
+
+
+def change_table_type_to_tuple(kwargs):
+    kwargs['table'] = (kwargs['table'], kwargs['separator'],
+                       kwargs['has_header'])
+
+
+def algos_options():
+    opts = create_and_fill_opts()
+
+    def decorator(func):
+        for opt_name, (opt_main_type, *opt_additional_types) in opts.items():
+
+            if opt_main_type == list:
+                click.option('--' + opt_name, multiple=True,
+                             type=opt_additional_types[0])(func)
+            elif opt_main_type == desb.Table:
+                click.option('--' + opt_name, type=str,
+                             required=True)(func)
+            else:
+                click.option('--' + opt_name, type=opt_main_type)(func)
+        return func
+
+    return decorator
+
+
+@click.command(add_help_option=False)
+@click.option('--help', is_flag=True,
+              callback=help_callback_func,
+              expose_value=False, is_eager=True)
+@click.option('--task', type=click.Choice(TASK_INFO.keys(),
+                                          case_sensitive=False), is_eager=True)
+@click.option('--algo', type=click.Choice(ALGOS.keys(),
```

Comment:
```suggestion
@click.option('--task', type=click.Choice([str(task) for task in  TASK_INFO.keys()],
                                          case_sensitive=False), is_eager=True)
@click.option('--algo', type=click.Choice([str(algo) for algo in ALGOS.keys()],
```
Иначе ошибка при неправильном параметре выглядит некрасиво.